engine/src/flutter/lib/ui/window/platform_configuration.h

```diff
@@ -461,6 +462,22 @@ class PlatformConfiguration final {
   ///
   void DispatchPointerDataPacket(const PointerDataPacket& packet);
 
+  //----------------------------------------------------------------------------
+  /// @brief      Requests from the engine if an embedded native view should
+  ///             accept touch at a given touch location.
+  ///
+  ///
+  /// @param[in]  view_id               The identifier of the flutter view that
+  ///                                   hosts the embedded view.
+  /// @param[in]  touch_began_location  The touch began location.
+  ///
+  /// @return     true if the embedded view should accept touch; false
+  /// otherwise.
+  ///
+  bool EmbeddedNativeViewShouldAcceptTouch(
+      int64_t view_id,
+      const flutter::PointData& touch_began_location);
```

Comment:
not a c++ expert, but do we want to pass by reference that may have memory complication? maybe use std::move.

Reply:
same for the entire drilling through shell

Reply:
I'd prefer the `double x` and `double y` approach here. We should pass this by value to avoid stack corruption.

We've had issues here before in split thread mode where the platform thread allocates a struct on the stack and then passes it down by reference to the UI thread. This results in stack corruption. But this could become a concern when we expose this API in the embedder API. We've been burned by stack corruption before with event channels.

Reply:
We synced offline and decided to pass by value (remove `&`).